### PR TITLE
Add implementation for join()

### PR DESCRIPTION
**Pull request by romaninsh**
Ported code from old DSQL.

todo - rethink alias syntax:

`join(‘address_user.id a’);`

### CODE REVIEW

**Comment by romaninsh** [comment]
Resolves ticket #14

File: docs/queries.rst [hunk 1]
```diff
@@ -319,6 +319,32 @@ You can also mix and match with expressions and strings::
...
+      Join results with additional table using "JOIN" statement in your query.
+
+      :param string $foreign_table: table to join (may include field and alias)
+      :param mixed  $master_field:  main field (and table) to join on or Expression
+      :param string $join_kind:     'left' (default), 'inner', etc - which type of join.
+
+When joinin with a different table, the results will be stacked by the SQL server
+so that fields from both tables are available.
+
+    $q->group('gender');
```
Comment (DarkSide666) [inline docs/queries.rst]: Why you're adding `group()` description here? `group` is not implemented yet.
Also `join` description looks quite small :)

File: docs/queries.rst [hunk 2]
```diff
@@ -319,6 +319,32 @@ You can also mix and match with expressions and strings::
 .. todo::
     strict mode
 
+Joining with other tables
+-------------------------
+
+  .. php:method:: join($field)
```
Comment (DarkSide666) [inline docs/queries.rst]: only `$field` parameter here?

File: src/Query.php [hunk 3]
```diff
@@ -317,6 +317,153 @@ protected function _render_from()
...
+     *  $q->join(array('a'=>'address', 'p'=>'portfolio'));
+     *
+     * You can use expression for more complex joins
+     *  $q->join('address',
+     *      $q->orExpr()
+     *          ->where('user.billing_id=address.id')
+     *          ->where('user.technical_id=address.id')
+     *  )
+     *
+     * @param string $foreign_table  Table to join with
```
Comment (DarkSide666) [inline src/Query.php]: `$foreign_table` can be also array

File: tests/QueryTest.php [hunk 4]
```diff
@@ -762,6 +762,36 @@ public function testOrder()
...
+            'left join `address` as `a` on `a`.`id` = `user`.`address_id`',
+            $this->q('[join]')->table('user')->join('address a')->render()
+        );
+        $this->assertEquals(
+            'left join `address` as `a` on `a`.`id` = `u`.`address_id`',
+            $this->q('[join]')->table('user','u')->join('address a')->render()
+        );
+        $this->assertEquals(
+            'left join `address` as `a` on `a`.`user_id` = `u`.`id`',
+            $this->q('[join]')->table('user','u')->join('address.user_id a')->render()
```
Comment (DarkSide666) [inline tests/QueryTest.php]: Setting joined table alias this way looks quite strange. It looks more like field alias in this syntax.

File: docs/queries.rst (hunk 1 above)
Comment (romaninsh) [inline docs/queries.rst]: maybe i mixed them up ;) it's #48

File: docs/queries.rst (hunk 2 above)
Comment (romaninsh) [inline docs/queries.rst]: oh sorry, i haven't finished.. l

**Pull request merged**